modules/rostests/apitests/iphlpapi/GetInterfaceName.c

```diff
@@ -0,0 +1,459 @@
+    ok(Error == ERROR_SUCCESS,
+       "GetLastError() returned %ld, expected ERROR_SUCCESS\n",
+       Error);
+    ok(ulOutBufLen == MAX_INTERFACE_NAME_LEN * 2, // Windows 7: (MAX_INTERFACE_NAME_LEN + 1) * 2
+       "ulOutBufLen is %ld, expected = MAX_INTERFACE_NAME_LEN * 2\n",
+       ulOutBufLen);
+    Error = wcslen(Name);
+    ok(Error == 0,
+       "wcslen(Name) is %ld, expected = 0\n",
+       Error);
+    if (Error > 0)
+        trace("Unexpected name: \"%S\"\n", Name);
+}
+
+static
+VOID
+test_GetInterfaceName(VOID)
+{
+    PIP_INTERFACE_INFO pInfo = NULL;
+    ULONG ulOutBufLen = 0;
+    DWORD ApiReturn;
+    WCHAR Name[MAX_ADAPTER_NAME];
+    UNICODE_STRING GuidString;
+    GUID AdapterGUID;
+    HINSTANCE hIpHlpApi;
+
+    ApiReturn = GetInterfaceInfo(pInfo, &ulOutBufLen);
+    ok(ApiReturn == ERROR_INSUFFICIENT_BUFFER,
+       "GetInterfaceInfo(pInfo, &ulOutBufLen) returned %ld, expected ERROR_INSUFFICIENT_BUFFER\n",
+       ApiReturn);
+    if (ApiReturn != ERROR_INSUFFICIENT_BUFFER)
+    {
+        skip("Can't determine size of IP_INTERFACE_INFO. Can't proceed\n");
+        return;
+    }
+
+    pInfo = (IP_INTERFACE_INFO *)HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, ulOutBufLen);
+    if (pInfo == NULL)
+    {
+        skip("pInfo is NULL. Can't proceed\n");
+        return;
+    }
+
+    ApiReturn = GetInterfaceInfo(pInfo, &ulOutBufLen);
+    ok(ApiReturn == NO_ERROR,
+        "GetInterfaceInfo(pInfo, &ulOutBufLen) returned %ld, expected NO_ERROR\n",
+        ApiReturn);
+    if (ApiReturn != NO_ERROR || ulOutBufLen == 0)
+    {
+        skip("GetInterfaceInfo failed with error %ld. Can't proceed\n", ApiReturn);
+        return;
+    }
+
+    if (pInfo->NumAdapters > 0)
+        CopyMemory(&Name, &pInfo->Adapter[0].Name, sizeof(Name));
+
+    if (pInfo->NumAdapters == 0)
+    {
+        HeapFree(GetProcessHeap(), 0, pInfo);
+        skip("pInfo->NumAdapters = 0. Can't proceed\n");
+        return;
+    }
+    trace("pInfo->NumAdapters: %lu\n", pInfo->NumAdapters);
+
+    HeapFree(GetProcessHeap(), 0, pInfo);
+
+    ApiReturn = wcsncmp(Name, L"\\DEVICE\\TCPIP_", 14);
+    ok(ApiReturn == 0,
+       "wcsncmp(Name, L\"\\DEVICE\\TCPIP_\", 14) returned %ld, expected 0\n",
+       ApiReturn);
+    if (ApiReturn != 0)
+    {
+        if (wcslen(Name) == 0)
+        {
+            skip("pInfo->Adapter[0].Name is empty. Can't proceed\n");
+            return;
+        }
+        else
+        {
+            // workaround for ReactOS
+            trace("pInfo->Adapter[0].Name = \"%ls\" is incorrect.\n", Name);
+            RtlInitUnicodeString(&GuidString, &Name[0]);
+        }
+    }
+    else
+    {
+        RtlInitUnicodeString(&GuidString, &Name[14]);
+    }
+
+    ApiReturn = RtlGUIDFromString(&GuidString, &AdapterGUID);
+    RtlFreeUnicodeString(&GuidString);
+    if (ApiReturn != 0)
+    {
+        skip("RtlGUIDFromString failed. Can't proceed\n");
+        return;
+    }
+
+    hIpHlpApi = GetModuleHandleW(L"iphlpapi.dll");
```

Comment:
LoadLibrary, it is not loaded anywhere in your test.
Now you are relying on another test to load the module for you, which might change at any time.

Reply:
No one test does load it, because it is loaded dynamically with the executable: [CMakeLists.txt](https://github.com/binarymaster/reactos/blob/000828290e34a4c12e006425fcf08d66866fc6c8/modules/rostests/apitests/iphlpapi/CMakeLists.txt#L12)

Reply:
Dynamically would be if one of the tests does LoadLibrary.
But basically, one of the tests links statically and this is why it works for you now.

Reply:
Well, it's not linked statically (in that case the lib would be inside test executable), but it's just added into imports table.